lib/pbench/agent/results.py

```diff
@@ -102,100 +122,78 @@ def make_result_tb(self):
                     tar.add(os.path.realpath(f))
         except tarfile.TarError:
             self.logger.error(
-                "tar ball creation failed for {}, skipping", self.result_dir
+                "Tar ball creation failed for {}, skipping", self.result_dir
             )
             try:
                 tarball.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
-            sys.exit(1)
-
-        self.make_md5sum(tarball)
+            except Exception as exc:
+                if not isinstance(exc, OSError) or exc.errno != errno.ENOENT:
+                    raise RuntimeError(
+                        f"Error removing failed tarball, '{tarball}', {exc}"
+                    )
 
         # The contract with the caller is to just return the full path to the
-        # created tar ball.
+        # created tarball.
         return str(tarball)
 
-    def make_md5sum(self, tarball):
-        tarball_md5 = Path(f"{tarball}.md5")
-        try:
-            hash_md5 = md5sum(tarball)
-            tarball_md5.write_text(f"{hash_md5} {tarball.name}\n")
-        except Exception:
-            self.logger.error("md5sum failed for {}, skipping", tarball)
-            try:
-                tarball.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
-            try:
-                tarball_md5.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error(
-                        "error removing failed tar ball MD5, {}", tarball_md5
-                    )
-            sys.exit(1)
-
 
 class CopyResultTb:
+    """CopyResultTb - Copies tarball from agent to server with the help of
+            RestAPI PUT request
+    """
+
     chunk_size = 4096
 
-    def __init__(self, controller, tarball, config, logger):
+    def __init__(
+        self, controller: str, tarball: str, config: PbenchAgentConfig, logger: Logger
+    ):
         self.tarball = Path(tarball)
         if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
-        self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
+            raise FileNotFoundError(f"Tarball does not exist, '{self.tarball}'")
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
+        server_rest_url = config.get("results", "server_rest_url")
         self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
 
-    def read_in_chunks(self, file_object):
+    def read_in_chunks(self, file_object: IO) -> Iterator[AnyStr]:
```

Comment:
`Iterator[AnyStr]` sounds odd, since it's binary tarball data rather than a string. Maybe `Iterator[Any]`? More literally it's a byte array, but I'm not entirely sure how it's typed formally.

Reply:
`AnyStr` is an umbrella for [both `str` and `bytes`](https://docs.python.org/3/library/typing.html#typing.AnyStr); I'll change it to `Iterator[bytes]`.